### DISCUSSION TITLE
Japanese Alphabetical Sorting for Manga Titles

### DISCUSSION THREAD

**Discussion by minahoku** [gotson/komga#1715] [Ideas]
I am a Japanese user. When arranging Japanese manga titles in Komga, the A-Z sorting becomes meaningless, as all titles end up under the "#" section. While I understand this feature was designed for English-speaking users, it would be greatly appreciated if you could implement a sorting option based on the Japanese syllabary (gojūon) for Japanese users.
https://en.wikipedia.org/wiki/Goj%C5%ABon

**Comment by gotson** [comment]
I checked the code, the API already returns groups of series by first character, so if there's other characters than A-Z, they still get returned. The logic is on the webui side, where the grouping needs to happen, especially for non A-Z characters.

We could imagine having different setup for other languages. There's some points that would need to be clarified though:
- the list of first characters would need to be stored somewhere. Would that kind of grouping depend on the user or on the server (where the admin would decide for everyone)?
- for Japanese, the sort title would need to be in hiragana. You would need to edit metadata to change any kanji / katakana into hiragana so they get grouped properly.

**Comment by minahoku** [comment]
In my setup, I've set all manga sort titles to hiragana, which I believe makes it easy to sort accordingly. Given this situation, I think it would be best if sorting were managed server-side (where the administrator decides for everyone). However, this is just my personal opinion, and I'm not sure how it would work in other users' environments.

**Comment by gotson** [comment]
@minahoku i would need some insights on how to properly display / handle this.

Here is an attempt:
<img width="1200" alt="image" src="https://github.com/user-attachments/assets/d333796a-5e73-4201-843e-5ec3d17291e6" />

For the dakuten and handakuten, how is it usually handled ? How about the katakana, should they be grouped in their hiragana equivalent ?

**Reply by minahoku** [reply]
In Japanese sorting, the order is "unvoiced sounds (e.g., 'ha' (は)) → voiced sounds (e.g., 'ba' (ば)) → semi-voiced sounds (e.g., 'pa' (ぱ))." Voiced and semi-voiced sounds are not displayed as separate lists but are included within the unvoiced sounds. (For example, "ga" (が) is included in the list for "ka" (か).) I believe it is reasonable to treat katakana the same way as hiragana.

**Reply by minahoku** [reply]
And, in Japanese, words that start with "ゃ", "ゅ", "ょ", "を", or "ん" generally do not exist, so always displaying them in a list might seem unnatural. However, since there is a slight possibility of their occurrence, such as in manga titles, it might be more natural to display them only when they exist. That said, this is not essential, so it can be treated as a low priority.

**Reply by gotson** [reply]
it would really help me if you could give me the grouping

**Reply by minahoku** [reply]
To convey this accurately, my reply will be a bit lengthy. I apologize for the inconvenience.


あ
い
う　う゛
え
お
か　が  
き　ぎ  
く　ぐ  
け　げ  
こ　ご  
さ　ざ  
し　じ  
す　ず  
せ　ぜ  
そ　ぞ  
た　だ  
ち　ぢ  
つ　づ  
て　で  
と　ど  
は　ば　ぱ  
ひ　び　ぴ  
ふ　ぶ　ぷ  
へ　べ　ぺ  
ほ　ぼ　ぽ  
ま  
み  
む  
め  
も  
や
ゆ
よ
ら  
り  
る  
れ  
ろ  
わ  
(を)  
(ん)

**Reply by minahoku** [reply]
![image](https://github.com/user-attachments/assets/028bbe23-d223-4728-a8a4-0e10f735f1ce)

**Reply by gotson** [reply]
so when displayed in a list it would look like this, does that make sense ?

<img width="1196" alt="image" src="https://github.com/user-attachments/assets/63332b1e-c6d6-4a0d-a574-e70e9202d575" />

**Reply by minahoku** [reply]
The sounds "が", "ぎ", "ぐ", "げ", "ご", "ざ", "じ", "ず", "ぜ", "ぞ", "だ", "ぢ", "づ", "で", "ど", "ば", "ぱ", "び", "ぴ", "ぶ", "ぷ", "べ", "ぺ", "ぼ", "ぽ" should not display buttons, and it is more natural to include "が" in the "か" button list, "ば" in the "は" button list, and "ぱ" in the "は" button list as well. For example, if there is a manga titled "はたらくさいぼう", "ぱられるぱらだいす", or "ばき", when you press the "は" button, they would appear in the order "はたらくさいぼう", "ばき", "ぱられるぱらだいす".

**Reply by gotson** [reply]
yes, i misread one of your comment.

Here is how it looks, with hiragana groupings shown, and it would filter both hiragana and katakana:
<img width="1188" alt="image" src="https://github.com/user-attachments/assets/44828a04-358d-478b-a4d7-3e915ecdb26d" />

<img width="1188" alt="image" src="https://github.com/user-attachments/assets/59aebd5c-621a-451f-9765-4393e3d9ce54" />

Including dakuten and handakuten:
<img width="1175" alt="image" src="https://github.com/user-attachments/assets/e244cc7c-f782-4a92-a6fa-8bfcfd48041e" />

Kanji would be grouped under `#`:
<img width="1193" alt="image" src="https://github.com/user-attachments/assets/dd635a65-1aa8-4e39-8154-93a5428f7930" />

**Reply by minahoku** [reply]
Wonderful! This is exactly what I was looking for! Titles that start with kanji, alphabet letters, or numbers will be classified under #, so I believe this is correct. I'm looking forward to the update.

**Reply by gotson** [reply]
i still need to find where to store that setting, but at least i know that for Japanese it's well implemented. Will be working on the configuration/storage next.

**Reply by gotson** [reply]
It will be in the next release

**Comment by gotson** [comment]
🎉 This issue has been resolved in `1.20.0` ([Release Notes](https://github.com/gotson/komga/releases/tag/1.20.0))
